cli/src/commands/squash.rs

```diff
@@ -213,6 +213,11 @@ from the source will be moved into the destination.
             diff_selector.select(&parent_tree, &source_tree, matcher, Some(&instructions))?;
         let selected_tree = tx.repo().store().get_root_tree(&selected_tree_id)?;
         let abandon = selected_tree.id() == source_tree.id();
+        if !abandon && selected_tree_id == parent_tree.id() {
```

Comment:
nit: better to add comment why `!abandon`?

I thought an empty commit shouldn't be squashed, but apparently some tests rely on this behavior to move commit metadata back to the parent commit.

Reply:
Hopefully clarified now. Looks good?

Reply:
Yes, thanks.

Btw, `jj squash -i` would previously error out in that situation. Just a comment, I don't think we'll want to add another special case. There's no way to distinguish "selected nothing" from "selected empty commit".